### PR TITLE
fix: honor custom{component}Image fields

### PR DESCRIPTION
**Pull request by chewong**
**Reason for Change**:

`customKubeAPIServerImage`, `customKubeControllerManagerImage`, `customKubeSchedulerImage`, and `customKubeProxyImage` have no effect after https://github.com/Azure/aks-engine/pull/2540 was merged. We should honor them when specified.

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/api/addons_test.go [hunk 1]
```diff
@@ -3131,7 +3131,7 @@ func TestSetAddonsConfig(t *testing.T) {
 						},
 					},
 				},
-			}, "1.15.4", "", common.KubernetesImageBaseTypeGCR),
+			}, "1.15.4", specConfig.KubernetesImageBase, common.KubernetesImageBaseTypeGCR),
```
Comment (chewong) [inline pkg/api/addons_test.go]: Fixed the following linting error:
```
==> Running Go linter <==
golangci-lint has version 1.23.7 built from 96964db on 2020-02-28T12:07:44Z
Running inside container
pkg/api/addons_test.go:4033:64: `overwriteDefaultAddons` - `kubernetesImageBase` always receives `""` (unparam)
func overwriteDefaultAddons(addons []KubernetesAddon, version, kubernetesImageBase, kubernetesImageBaseType string) []KubernetesAdd
```

File: pkg/api/addons_test.go [hunk 2]
```diff
@@ -3991,14 +4030,14 @@ func concatenateDefaultAddons(addons []KubernetesAddon, version string) []Kubern
 	return defaults
 }
 
-func overwriteDefaultAddons(addons []KubernetesAddon, version, kubernetesImageBase, kubernetesImageBaseType string) []KubernetesAddon {
+func overwriteDefaultAddons(addons []KubernetesAddon, version string) []KubernetesAddon {
```
Comment (chewong) [inline pkg/api/addons_test.go]: Fixed the following linting error: 
``` 
==> Running Go linter <== 
golangci-lint has version 1.23.7 built from 96964db on 2020-02-28T12:07:44Z
Running inside container
pkg/api/addons_test.go:4033:64: `overwriteDefaultAddons` - `kubernetesImageBase` always receives `""` (unparam) func overwriteDefaultAddons(addons []KubernetesAddon, version, kubernetesImageBase, kubernetesImageBaseType string) [] KubernetesAddon
```

**Comment by chewong** [comment]
@jackfrancis could you take another look at this?

File: pkg/api/addons_test.go (hunk 2 above)
Comment (jackfrancis) [reply pkg/api/addons_test.go]: booby trap!

**Pull request merged**